test/XliffSelect.test.js

```diff
@@ -589,4 +589,101 @@ describe("xliff select translation units", function() {
         '</xliff>';
         expect(actual).toBe(expected);
     });
+
+    test("Select while avoiding duplicated file names", function() {
+        expect.assertions(2);
+
+        var settings = {
+            xliffVersion: 2,
+            infiles: [
+                "test/testfiles/xliff20/app1/en-US.xliff", // should only read this file once
+                "test/testfiles/xliff20/app1/en-US.xliff",
+                "test/testfiles/xliff20/app1/en-US.xliff",
+                "test/testfiles/xliff20/app1/en-US.xliff",
+                "test/testfiles/xliff20/app1/en-US.xliff"
+            ]
+        };
+
+        var target = XliffSelect(settings);
+        expect(target).toBeTruthy();
```

Comment:
This assertion would be truthy also if the `target` is an empty object.

If this is not okay, then you could do 
`expect(target).toBeInstanceOf(Xliff);`
instead. Then for an empty object this assertion would fail.

Reply:
Those sort of expects are pre-conditions actually. The real test is the one where actual = expected. I just wanted to make sure that the XliffSelect produced something at all before we deference it on line 610.

Reply:
if you want to
> make sure that the XliffSelect produced something

then I'd go with
`expect(target).toBeInstanceOf(Xliff);`